source/elements/oneTBB/source/named_requirements/algorithms/par_for_each_body.rst

```diff
@@ -30,18 +30,31 @@ It should also meet one of the following requirements:
     ``ItemType`` may be optionally passed to ``Body::operator()`` by reference.
     ``const`` and ``volatile`` type qualifiers are also applicable.
 
-ItemType
---------
-
-The argument type ``ItemType`` should either satisfy the *CopyConstructibe*, *MoveConstructibe* or
-both requirements from the ISO C++ [utility.arg.requirements] section.
-If the type is not *CopyConstructibe*, there are additional usage restrictions:
-
-* If ``Body::operator()`` accepts an argument by value, or if the ``InputIterator`` type
-  from :doc:`parallel_for_each algorithm <../../algorithms/functions/parallel_for_each_func>` does
-  not satisfy the `Forward Iterator` requirements from the [forward.iterators] ISO C++ Standard section,
-  dereferencing an ``InputIterator`` must produce an rvalue reference.
-* Additional work items should be passed to the feeder as rvalues, for example, via the ``std::move`` function.
+Terms
+-----
+
+* ``iterator`` determines the type of the iterator passed into the ``parallel_for_each`` algorithm,
+  which is ``decltype(std::begin(c))`` for the overloads that accept the ``Container`` template argument or ``InputIterator``.
+* ``value_type`` - the type ``std::iterator_traits<iterator>::value_type``.
+* ``reference`` -  the type ``std::iterator_traits<iterator>::reference``.
+
+``oneapi::tbb::parallel_for_each`` requires the ``Body::operator()`` call with an object of the ``reference`` type to be well-formed if
+the ``iterator`` meets all of the `Forward iterator` requirements described in the [forward.iterators] section of the 
+ISO C++ Standard.
+
+`oneapi::tbb::parallel_for_each algorithm <../../algorithms/functions/parallel_for_each_func>`
+requires the ``Body::operator()`` call with an object of type ``const value_type&`` or ``value_type&&`` to be well-formed if:
```

Comment:
```suggestion
requires the ``Body::operator()`` call with an object of type ``const value_type&`` or ``value_type&&`` to be well-formed if all of the following requirements are fulfilled:
```

Reply:
I'd rather say **requirements are met**, cause it's more impersonal

Reply:
```suggestion
requires the ``Body::operator()`` call with an object of type ``const value_type&`` or ``value_type&&`` to be well-formed if all of the following requirements are met:
```